### PR TITLE
refactor(service-client): Make service client easy to mock

### PR DESCRIPTION
**Pull request by vinagesh**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

## Description of the changes
Following the same pattern as JobClient, making a change for ServiceClient to make it more mockable.

### CODE REVIEW

**Review by timtay-microsoft — APPROVED** [review]

File: iothub/service/src/ServiceClient.cs [hunk 1]
```diff
@@ -95,55 +150,184 @@ public static ServiceClient CreateFromConnectionString(string connectionString,
...
+               ExceptionHandlingHelper.GetDefaultErrorMapping(),
+               s_defaultOperationTimeout,
+               transportSettings.HttpProxy);
+            var serviceClient = new ServiceClient(iotHubConnection, iotHubConnectionString.IotHubName, httpClientHelper, options);
             return serviceClient;
         }
 
         /// <summary>
-        /// Open the ServiceClient instance.
+        /// Open the ServiceClient instance. This call is executed over AMQP.
```
Comment (drwill-ms) [inline iothub/service/src/ServiceClient.cs:165]: We should avoid using the word 'executed' for the associated negative connotations. How about 'This call is made over AMQP.'?

File: iothub/service/tests/ConnectionString/ServiceClientConnectionStringTests.cs [hunk 2]
```diff
@@ -26,7 +26,7 @@ public virtual IotHubConnectionStringBuilder Populate(IotHubConnectionStringBuil
         public void ServiceClientConnectionStringDefaultScopeDefaultCredentialTypeTest()
         {
             string connectionString = "HostName=acme.azure-devices.net;SharedAccessKeyName=AllAccessKey;SharedAccessKey=dGVzdFN0cmluZzE=";
-            var serviceClient = (AmqpServiceClient)ServiceClient.CreateFromConnectionString(connectionString);
+            var serviceClient = (ServiceClient)ServiceClient.CreateFromConnectionString(connectionString);
```
Comment (drwill-ms) [inline iothub/service/tests/ConnectionString/ServiceClientConnectionStringTests.cs:29]: Does it still require casting now?

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/tests/ConnectionString/ServiceClientConnectionStringTests.cs (hunk 2 above)
Comment (vinagesh) [reply iothub/service/tests/ConnectionString/ServiceClientConnectionStringTests.cs:29]: Good point. It should not. I will double check and update accordingly.

**Pull request merged**